echo360/course.py

```
@@ -192,26 +201,114 @@ def course_name(self):
+            course_data = self._fetch_course_data_via_json_api()
+            if course_data.get("data"):
+                self.course_data = course_data
+                return self.course_data
             _LOGGER.debug(
-                "Dumping course page at %s: %s",
-                self.video_url,
-                self._driver.page_source,
+                "JSON syllabus endpoint returned no lessons; falling back to "
+                "scraping the section home page"
+            )
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)